File: frontend/src/fakeBackend/services/ClientService.js
import { db } from "../db";
import { generateId } from "../util/generateID";

export const ClientService = {
  getClient(clientId) {
    return db.clients.find(c => c.id === clientId);
  },

  getAllClients() {
    return db.clients.map(client => {
      const locations = db.locations.filter(loc => loc.clientId === client.id);
      return {
        ...client,
        locationCount: locations.length,
        locations: locations
      };
    });
  },

  addClient(name) {
    const client = {
      id: generateId(),
      name
    };
    db.clients.push(client);
    return client;
  }
};

File: backend/src/main/java/com/emission_impossible/carbon_modelling_tool/service/ClientService.java
package com.emission_impossible.carbon_modelling_tool.service;

import com.emission_impossible.carbon_modelling_tool.model.Client;
import com.emission_impossible.carbon_modelling_tool.repository.ClientRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ClientService {

    private final ClientRepository clientRepository;

    public ClientService(ClientRepository clientRepository){
        this.clientRepository = clientRepository;
    }

    public Client getClient(Long clientId){
        return this.clientRepository.findById(clientId)
                .orElseThrow(() -> new RuntimeException(String.format("No client found for client id %d", clientId)));
    }

    public List<Client> getAllClients(){
        return this.clientRepository.findAll();
    }

    public Client addClient(String name){

        Client newClient = new Client(name);

        this.clientRepository.save(newClient);

        return newClient;
    }


}
